stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2628,196 +2658,156 @@ final internal class _Native${Self}StorageImpl<${TypeParameters}> {
-public // @testable
-struct _Native${Self}Storage<${TypeParametersDecl}> :
-  _HashStorage, CustomStringConvertible {
-  internal typealias Owner = _Native${Self}StorageOwner<${TypeParameters}>
-  internal typealias StorageImpl = _Native${Self}StorageImpl<${TypeParameters}>
+// TODO: conditionally conform to _HashStorage for better compile-time safety.
+// TODO: conditionally conform to CustomStringConvertible.
+extension _Native${Self}Storage 
+  where ${'Key' if Self == 'Dictionary' else 'Element'} : Hashable
+{
   internal typealias SequenceElement = ${Sequence}
-%if Self == 'Set':
-  internal typealias SequenceElementWithoutLabels = Element
-%else:
-  internal typealias SequenceElementWithoutLabels = (Key, Value)
-%end
-  internal typealias Storage = _Native${Self}Storage<${TypeParameters}>
-
-%if Self == 'Set': # Set needs these to keep signatures simple.
-  internal typealias Key = ${TypeParameters}
-  internal typealias Value = ${TypeParameters}
-%end
-
-  internal let buffer: StorageImpl
-
-  internal let initializedEntries: _UnsafeBitMap
-  internal let keys: UnsafeMutablePointer<Key>
-%if Self == 'Dictionary':
-  internal let values: UnsafeMutablePointer<Value>
-%end
-
-  internal init(capacity: Int) {
-    buffer = StorageImpl.create(capacity: capacity)
-    initializedEntries = _UnsafeBitMap(
-      storage: buffer._initializedHashtableEntriesBitMapStorage,
-      bitCount: capacity)
-    keys = buffer._keys
-%if Self == 'Dictionary':
-    values = buffer._values
-%end
-    _fixLifetime(buffer)
-  }
 
-  internal init(minimumCapacity: Int = 2) {
-    // Make sure there's a representable power of 2 >= minimumCapacity
-    _sanityCheck(minimumCapacity <= (Int.max >> 1) + 1)
-
-    var capacity = 2
-    while capacity < minimumCapacity {
-      capacity <<= 1
-    }
-
-    self = _Native${Self}Storage(capacity: capacity)
-  }
-
-  @_transparent
-  public // @testable
-  var capacity: Int {
-    return buffer._capacity
+#if _runtime(_ObjC)
+  func bridgingStorage() -> _Native${Self}BridgingStorage<${TypeParameters}> {
+    return _Native${Self}BridgingStorage(nativeStorage: self)
   }
+#endif
 
-  @_versioned
-  @_transparent
-  internal var count: Int {
-    get {
-      return buffer._count
-    }
-    nonmutating set(newValue) {
-      buffer._count = newValue
+  /// A textual representation of `self`.
+  var description: String {
+    var result = ""
+#if INTERNAL_CHECKS_ENABLED
+    for i in 0..<_capacity {
+      if isInitializedEntry(at: i) {
+        let key = self.key(at: i)
+        result += "bucket \(i), ideal bucket = \(_bucket(key)), key = \(key)\n"
+      } else {
+        result += "bucket \(i), empty\n"
+      }
     }
-  }
-
-  @_transparent
-  internal var maxLoadFactorInverse: Double {
-    return buffer._maxLoadFactorInverse
+#endif
+    return result
   }
 
   @_versioned
   @inline(__always)
   internal func key(at i: Int) -> Key {
-    _sanityCheck(i >= 0 && i < capacity)
+    _sanityCheck(i >= 0 && i < _capacity)
```

Comment:
Yes, I believe the old assertion was in error (and always too large, because it was in terms of bytes). With the new design it's impossible to get this wrong; _capacity is right.